### PR TITLE
[4051] - scandiPWA version in footer is hardcoded - fixed

### LINKED ISSUE #4051: ScandiPWA version in admin footer appears only if the theme is called /scandipwa
**Description**:
The ScandiPWA version in the admin footer appears only if the theme is called scandipwa/scandipwa.

The issue is here: https://github.com/scandipwa/customization/blob/1.5.9/src/Block/Page/Footer.php#L26. 
It's hardcoded to only work if the theme is called `scandipwa/scandipwa`. 
So it doesn't show in other cases. For eg. In readymage we are using `scandipwa/theme` and for this [CSA command](https://docs.create-scandipwa-app.com/getting-started/getting-started#npx) it's `scandipwa/my-app`.

**Expected behavior**:
The version used appears in the footer of the admin panel.

**Screenshots, Video, Logs**
image

**To Reproduce**:
1. Go to the admin
2. Scroll down to the footer
3. If you do it in Rmg, it won't appear.

**Additional context**:
Suggestion from Ričards Rikmanis: "it would be look up based on `frontend/scandipwa/*`, this way we can account on any custom name. In cases there are multiple themes that start with `frontend/scandipwa/` pick the first one and thats it."

### PR DESCRIPTION
**Pull request by boltzmannsBane**
**Related issue(s):**
* Fixes scandipwa/scandipwa/issues/4051

**Problem:**
* The theme version in the footer only displays if the theme is called a particular way

**In this PR:**
* Fetching the applied theme path from the db
* Checking if the applied theme has a package.json with scandipwa dependency
* If yes, display its version. If not, display "not selected"

### CODE REVIEW

File: src/Block/Page/Footer.php [hunk 1]
```diff
@@ -83,9 +87,9 @@ public function getPackageJsonData() {
 
         if (file_exists($pathToTheme)) {
             $packageData = json_decode(file_get_contents($pathToTheme), true);
-            $this->scandiPWAPackgeVersion = $this->getScandiPWAFromPackageData($packageData);
+             $this->scandiPWAPackgeVersion = $this->getScandiPWAFromPackageData($packageData);
```
Comment (zans-laksa) [inline src/Block/Page/Footer.php]: Please remove the space.

File: src/Block/Page/Footer.php [hunk 2]
```diff
@@ -83,9 +87,9 @@ public function getPackageJsonData() {
...
         } else {
-            $this->scandiPWAPackgeVersion = false;
+            $this->scandiPWAPackgeVersion = "not selected";
```
Comment (zans-laksa) [inline src/Block/Page/Footer.php]: Please change the label to `n/a`.

File: src/Block/Page/Footer.php [hunk 3]
```diff
@@ -98,8 +102,16 @@ public function getPackageJsonData() {
...
         $themeDirectoryPath = null;
 
+        $connection = $this->resourceConnection->getConnection();
+        // fetch the id of the currently applied theme
+        $themeid_query = "SELECT value FROM core_config_data WHERE config_id = 37";
+        $themeid = $connection->fetchAll($themeid_query);
+        // use this id to fetch the applied theme itself
+        $applied_theme_query = "SELECT * FROM theme WHERE theme_id = " . $themeid[0]['value'];
+        $applied_theme = $connection->fetchAll($applied_theme_query);
+
```
Comment (zans-laksa) [inline src/Block/Page/Footer.php]: This is exactly the opposite of what we were talking about.
No database queries. Only going through the directory list and checking files.

File: src/Block/Page/Footer.php [hunk 4]
```diff
@@ -98,8 +102,16 @@ public function getPackageJsonData() {
     public function getScandiPWADirectoryPath() {
         $themeDirectoryPath = null;
 
+        $connection = $this->resourceConnection->getConnection();
+        // fetch the id of the currently applied theme
+        $themeid_query = "SELECT value FROM core_config_data WHERE config_id = 37";
```
Comment (zans-laksa) [inline src/Block/Page/Footer.php]: Why did you hardcode the id, if it will be different for every project?

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: src/Block/Page/Footer.php [hunk 5]
```diff
@@ -85,7 +80,7 @@ public function getPackageJsonData() {
             $packageData = json_decode(file_get_contents($pathToTheme), true);
             $this->scandiPWAPackgeVersion = $this->getScandiPWAFromPackageData($packageData);
         } else {
-            $this->scandiPWAPackgeVersion = false;
+            $this->scandiPWAPackgeVersion = "n/a";
```
Comment (zans-laksa) [inline src/Block/Page/Footer.php]: Please use single quotes for strings.

File: src/Block/Page/Footer.php [hunk 6]
```diff
@@ -99,7 +94,7 @@ public function getScandiPWADirectoryPath() {
         $themeDirectoryPath = null;
 
         foreach ($this->themeList as $theme) {
-            if ($theme->getFullPath() === self::SCANDIPWA_COMPONENT_NAME) {
+            if (str_contains($theme->getFullPath(), "frontend/scandipwa")) {
```
Comment (zans-laksa) [inline src/Block/Page/Footer.php]: Please use single quotes for strings.

Comment (zans-laksa) [reply src/Block/Page/Footer.php]: Please move `frontend/scandipwa` to the class constant.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: src/Block/Page/Footer.php (hunk 6 above)
Comment (boltzmannsBane) [reply src/Block/Page/Footer.php]: done

**Review by zans-laksa — APPROVED** [review]

**Pull request merged**